### PR TITLE
[FLINK-25676][python] Support set_description in DataStream API

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request add support of set_description in DataStream API*

## Verifying this change

This change is a trivial rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-python/pyflink/datastream/data_stream.py [hunk 1]
```diff
@@ -231,6 +231,24 @@ def slot_sharing_group(self, slot_sharing_group: Union[str, SlotSharingGroup]) -
             self._j_data_stream.slotSharingGroup(slot_sharing_group)
         return self
 
+    def set_description(self, description: str):
```
Comment (HuangXingBo) [inline flink-python/pyflink/datastream/data_stream.py:234]:
```suggestion
    def set_description(self, description: str) -> 'DataStream':
```

File: flink-python/pyflink/datastream/data_stream.py [hunk 2]
```diff
@@ -231,6 +231,24 @@ def slot_sharing_group(self, slot_sharing_group: Union[str, SlotSharingGroup]) -
...
+        Sets the description for this operator.
+
+        Description is used in json plan and web ui, but not in logging and metrics where only
+        name is available. Description is expected to provide detailed information about the
+        operator, while name is expected to be more simple, providing summary information only,
+        so that we can have more user-friendly logging messages and metric tags without losing
+        useful messages for debugging.
+
+        :param description: The description for this operator.
+        :return: The sink with new description.
```
Comment (HuangXingBo) [inline flink-python/pyflink/datastream/data_stream.py:245]:
```suggestion
        :return: The operation with new description.
```

**Review by HuangXingBo — APPROVED** [review]
Thanks a lot for the PR. Looks good overall. I only left two minor comments.

**Comment by dianfu** [comment]
@HuangXingBo Thanks for the review. Will address the comments during merging the PR.

**Pull request closed without merging**